client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx

```diff
@@ -99,14 +99,11 @@ export default function WPCheckoutOrderReview( {
 	const cartKey = useCartKey();
 	const { responseCart, removeCoupon, couponStatus } = useShoppingCart( cartKey );
 	const isPurchaseFree = responseCart.total_cost_integer === 0;
-	const isDesktop = useViewportMatch( 'mobile', '>=' );
 	const reduxDispatch = useDispatch();
 
 	useEffect( () => {
-		const experimentCheck = isDesktop
-			? 'registration_email_only_desktop_random_usernames'
-			: 'registration_email_only_mobile_random_usernames';
-		let shouldCheck = true;
+		const experimentCheck = retrieveExperimentName();
+		let shouldCheck = experimentCheck?.length > 0;
```

Comment:
TS error: client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx:106:21 - error TS2532: Object is possibly 'undefined'.

Reply:
The problem here is that `undefined > 0` is not an expression TS likes. There's various ways you can resolve this; here's one:

```suggestion
		let shouldCheck = experimentCheck ? experimentCheck.length > 0 : false;
```